### PR TITLE
Feature/search bar

### PR DESCRIPTION
**Pull request by bluebill1049**
- [x] style the search section.

### CODE REVIEW

File: src/components/Nav.tsx [hunk 1]
```diff
@@ -26,98 +28,105 @@ export default function Nav({ defaultLang }: { defaultLang: string }) {
...
+  React.useEffect(() => {
+    if (setting.isFocusOnSearch && window.innerWidth < 767) {
+      setLang(false)
+    } else if (!setting.isFocusOnSearch) {
+      setLang(true)
+    }
+
+    if (
+      setting.isFocusOnSearch &&
+      window.innerWidth < 767 &&
```
Comment (pmaier983) [inline src/components/Nav.tsx:40]: i think in the future we should have a breakpoint object somewhere. Each of the values mapping to a normal device (iphone, normal monitor and the such) (@TechDebt)
```javascript
const breakpoints = {
	xsm: 500,
	sm: 756,
	m: 1000,
	l: 1500,
	xl: 2000,
}
```

File: src/components/layout.tsx [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import * as React from "react"
 import { Animate } from "react-simple-animate"
-import Prism from "prismjs"
```
Comment (pmaier983) [inline src/components/layout.tsx:3]: I need to configure our linter so that it catches these!

File: src/components/Search.tsx [hunk 3]
```diff
@@ -1,17 +1,65 @@
...
+import * as React from "react"
+import { useStateMachine } from "little-state-machine"
+import { updateSetting } from "../actions/settingActions"
 import searchStyles from "./Search.module.css"
 
 const Search = () => {
+  const timer = React.useRef<any>({})
+  const { action, state } = useStateMachine(updateSetting)
+
+  React.useEffect(() => {
```
Comment (pmaier983) [inline src/components/Search.tsx:10]: We need to pick gatsby config or this. I don't think both is necessary.

File: gatsby-config.js [hunk 4]
```diff
@@ -70,7 +70,7 @@ module.exports = {
       options: {
         apiKey: "953c771d83fb6ffd55fe58da997f2d9d",
         indexName: "react-hook-form",
-        inputSelector: "#algolia-doc-search",
+        inputSelector: "#fakeSearch",
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply gatsby-config.js:73]: the gatsybe thing doesn't work... it's targeting a fark searh, take a look the search component.

File: src/components/Search.tsx (hunk 3 above)
Comment (bluebill1049) [reply src/components/Search.tsx:10]: gatsbye one doesn't work when you navigate around the page, try it.

File: src/components/layout.tsx (hunk 2 above)
Comment (bluebill1049) [reply src/components/layout.tsx:3]: oh i see. will roll it back.

**Comment by bluebill1049** [comment]
> I really love the design here! I think we still need to work on implementation a bit, as on my computer the deploy is a bit janky.
> 
> image
> 
> In the image above you can click on either bubble and it will open up a different search.
> The magnifying glass is not centered on the first bubble, and the second bubble has nothing in it.
> 
> Overall awesome idea for the design Bill! If you need any help ping me 😄 (you must have your hands full with both this and bugs!)

going to fix that, doesn't looks like in local 👍

**Pull request merged**